script/deploy/deploySingleContract.sh

```
@@ -363,16 +363,27 @@ deploySingleContract() {
   fi
 
   # extract constructor arguments from return data
-  # Use sed to handle multi-line JSON (critical for large hex strings that cause forge to output multi-line JSON)
-  local JSON_DATA
-  JSON_DATA=$(echo "$RAW_RETURN_DATA" | sed -n '/{"logs":/,/}$/p' | tr -d '\n' | sed 's/} *$/}/')
-  
-  # Fallback: if sed method fails, try grep method (but this may truncate multi-line JSON)
-  if [[ -z "$JSON_DATA" || "$JSON_DATA" == "" ]]; then
-    JSON_DATA=$(echo "$RAW_RETURN_DATA" | grep -o '{"logs":.*}' | tail -1)
+  # Prefer broadcast artifact (reliable); fall back to forge stdout (can contain newlines/stray "0x" in value)
+  local BROADCAST_JSON="broadcast/$(basename "$FULL_SCRIPT_PATH")/$(getChainId "$NETWORK")/run-latest.json"
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)